src/lib.rs

```diff
@@ -255,6 +279,25 @@ pub fn encode_config_buf(input: &[u8], config: Config, buf: &mut String) {
             raw.push(0x3d);
         }
     }
+
+    if config.line_size.is_some() && config.line_size.unwrap() > 0 {
+        let line_size = config.line_size.unwrap();
+        let len = raw.len();
+        let mut i = 0;
+        let mut j = 0;
+
+        while i < len {
+            if i > 0 && i % line_size == 0 {
+                match config.line_ending {
+                    LineEnding::LF => { raw.insert(j, b'\n'); j += 1; }
+                    LineEnding::CRLF => { raw.insert(j, b'\r'); raw.insert(j + 1, b'\n'); j += 2; }
```

Comment:
ya it would be. alternative simple solution could just allocate another vec and copy in linear